### ISSUE TITLE
RFC: Swap x/net/websocket with gorilla/websocket

### ISSUE THREAD

**Issue by bitwalker** [convox/rack#231]
I'm working on swapping the websocket handling to use the gorilla implementation, and was wondering if you'd be interested in a PR which makes that change. One of the motivations behind it (for me) is so that I can use keepalives to keep the logs streaming until the client closes the connection, amongst other things.

If you have a particular reason you chose not to use that lib, I'd love to know, but if you're interested in the code once it's ready, I'd be glad to upstream it.

**Comment by nzoschke** [comment]
Thank you for the question, and your patience over the holidays.

The history indicates we did use gorilla/websocket and removed it, however I can't pinpoint a specific reason why.

I have a slight preference for the x/ libraries for future compatibility, but would happily review a patch that re-introduces gorilla/websocket if it improves log streaming.

Can you offer steps to reproduce a problem with the log streaming and the x/net/websocket library?

**Comment by bitwalker** [comment]
Hey thanks for the quick response! Looks like the websocket issues we ran into (connection was dropped after ~30s) is fixed, so my PR would be mostly irrelevant anyway. I was seeing a lot of people recommending using gorilla/websocket over x/net for a variety of reasons, and figured since you were already using gorilla libs that it would make sense to use their websocket lib too..but at the moment I don't see any reason to change what's in place today since my primary annoyance has been addressed. I'll close this for now :)
